### DISCUSSION TITLE
Theme Override for admin_template.php in core/templates

### DISCUSSION THREAD

**Discussion by Alex-e107nl** [e107inc/e107#5369] [General]
Is it possible to release this for the admin theme? The code in admin_template.php is very outdated and I had to rewrite it completely to use it with BS 5..

**Comment by Alex-e107nl** [comment]
@Jimmi08 isnt there a hidden key somewhere that i can use so the admin_template.php can be used from the admin theme folder? Just like with the auth.php? Or maybe is it possible to use theme_shortcodes.php to override things like:

```
$ADMIN_TEMPLATE['nav']['button'] = '
	<li class="nav-item dropdown">
		<a class="nav-link dropdown-toggle caret-off navbar-admin-button"  role="button" data-toggle="dropdown" data-bs-toggle="dropdown" data-target="#" href="{LINK_URL}" title="{LINK_DESCRIPTION}">
		 {LINK_IMAGE}
		{LINK_TEXT}
		</a> 
		{SUB_MENU}
	</li>
';
```
Now i can use bs3 and bs5 as an admin theme but i need to switch the admin_template.php also....

**Comment by Jimmi08** [comment]
@Alex-e107nl 
It is not allowed by default.
**One way**
Go to core admin_shortcodes.php , find `$template = e107::getCoreTemplate('admin', 'nav', false);`  and change last parameter to true 
`$template = e107::getCoreTemplate('admin', 'nav', true);`

This part is responsible for navigation. 
For other parts of admin you need to do the same thing in admin/header.php 

If it works, ask them to change this - you can't replace admin icons template either.  I think it is something forgotten, not intention. 

**Other way - without changing core files**
Not tested, because I don't have bootstrap 5 admin theme now
Go to core/templates/bootstrap4   or create bootstrap5 folder
and put your templates there. It should work. 

There is callback if bootstrap5 version doesn't exist, it looks in boostrap4 folder (I dislike this because it is confusing - you have BT 5 and you need to look in bootstrap4 folder). I prefer to create bootstrap5 folder if you don't mind.

 If you want to test this, go to e107_class.php and look for coreTemplatePath() method. Your $id for admin area is "admin".

**Reply by Alex-e107nl** [reply]
THX @Jimmi08 as always, the Other way - without changing core files is working fine. But for a definitive solution I have to adjust the files, then I can do a pull and keep my fingers crossed that 2025 will be the year of e107 with a bootstrap 5 admin :-) 

It is very strange to see that if you forget a TRUE somewhere your own theme is suddenly filled with BS3 code.. I haven't found them all yet so...

Another step further and I know that one of the conditions of e107 is the KISS principle so I try to hold on to that as best as possible.... (Keep It Simple Stupid or is it Keep It Stupid Simple :-)

**Reply by Alex-e107nl** [reply]
Oke,
in the admin_template.php is a mention:
`NEW ADMIN MENU TEMPLATE  * see function e107::getNav()->admin() in e107_admin/header.php`
That is the only thing that is not read from my admin_template.php file now, the rest works!

When i look in the e107_admin/header.php` there is a mention // function e_admin_me/nu moved to boot.php (e107::getNav()->admin)
So off we go to boot.php ..

What i see is a mention about the admin icons:
// Get Icon constants, theme override (theme/templates/admin_icons_template.php) is allowed ?

The code with e107::getNav()->admin() = 

```
if(!defset('e_ADMIN_UI') && !defset('e_PAGETITLE'))
{
	e107::getDebug()->logTime('[boot.php: Loading adminLinks(\'legacy\')]');
	$array_functions = e107::getNav()->adminLinks('legacy'); // replacement see e107_handlers/sitelinks.php
	foreach($array_functions as $val)
	{
	    $link = str_replace("../","",$val[0]);
		//if(strpos(e_SELF,$link)!==FALSE)
	//	{
	 //   	define('e_PAGETITLE',$val[1]);
	//	}
	}
}
```
There's also a mention: DEPRECATED - use e_adm/in_menu()  e107::getNav()->admin

**So now i am lost...** 

The changes i have done to get the rest working is:
```
header.php
578 - 579
$ADMIN_HEADER = e107::getCoreTemplate('admin', 'header', true);
$ADMIN_MODAL = e107::getCoreTemplate('admin', 'modal', true);
```

```
admin_shortcodes.php
2138
$template = e107::getCoreTemplate('admin', 'nav', true);
```

Can someone help please? I am stil trying to get my admin_template.php working from the admin theme, that is the only thing now to make it possible to switch between bootstrap3 and the new bootstrap5 theme for the admin...

Without it the links look like:	
![links_without](https://github.com/user-attachments/assets/423b5fa3-0f36-4d49-9e7b-89c9816f6fe9)

And when its working they look like:
![links_with](https://github.com/user-attachments/assets/38c34227-b05c-4d82-a78c-640cb7fef445)

**Reply by Alex-e107nl** [reply]
Ok, other solution then until it is solved: 
```

if (copy($source, $dest)) {
    /* echo 'File copied successfully '; */
```

Wishing everyone a happy and healthy 2025!
